glean-core/src/metrics/labeled.rs

```diff
@@ -32,8 +39,36 @@ const MAX_LABEL_LENGTH: usize = 61;
 /// * `1.not_fine`
 /// * `this.$isnotfine`
 /// * `-.not_fine`
-static LABEL_REGEX: Lazy<Regex> =
-    Lazy::new(|| Regex::new("^[a-z_][a-z0-9_-]{0,29}(\\.[a-z_][a-z0-9_-]{0,29})*$").unwrap());
+fn matches_label_regex(value: &str) -> bool {
+    let mut iter = value.chars();
+
+    loop {
+        // Match the first letter in the word.
+        match iter.next() {
+            Some('_') | Some('a'..='z') => (),
+            _ => return false,
+        };
+
+        // Match subsequent letters in the word.
+        let mut count = 0;
+        loop {
+            match iter.next() {
+                // We are done, so the whole expression is valid.
+                None => return true,
+                // Valid characters.
+                Some('_') | Some('-') | Some('a'..='z') | Some('0'..='9') => (),
+                // We ended a word, so iterate over the outer loop again.
+                Some('.') => break,
+                // An invalid character
+                _ => return false,
+            }
+            count += 1;
+            if count == 29 {
```

Comment:
Should this be if count == 30 since we start counting at 0 and allow for 30 characters?

Reply:
The first character (since it's handled differently) is outside of the loop, so there's only 29 of these characters allowed.  Also see regex itself in the comment above.  Worth mentioning in a comment, for sure.

Reply:
Nevermind, you already peeled off the first character.

Reply:
jinx :)